sites/svelte-5-preview/src/routes/docs/content/01-api/02-runes.md

```diff
@@ -241,6 +241,45 @@ let { a, b, c, ...everythingElse } = $props<MyProps>();
 
 Note that you can still use `export const` and `export function` to expose things to users of your component (if they're using `bind:this`, for example).
 
+## `$inspect`
+
+The `$inspect` rune is roughly equivalent to `console.log`, with the exception that it will re-run whenever the
+arguments change. `$inspect` tracks reactive state deeply, meaning that updating something inside an object
+or array using [fine-grained reactivity](/docs/fine-grained-reactivity) will cause it to re-fire.
+
+```svelte
+<script>
+	let count = $state(0);
+	let message = $state(0);
+
+	$inspect({ count, message }); // will console.log when `count` or `message` change
+</script>
+
+<button onclick={() => count++}>Increment</button>
+<input bind:value={message} />
+```
+
+If the last argument is a function, it will be invoked instead of `console.log`. The first argument is `changed` —
```

Comment:
the counterpoint i suppose is that for people not using the callback, it's nice to be able to do this sort of thing

```js
$inspect('>>> HERE', foo);
```

Reply:
@trueadm made the excellent observation that you might need to log a function at some point, in which case it would be ambiguous. I've updated the signature:

```js
$inspect({ ...stuff });

$inspect({ ...stuff }), (stuff, type) => {
  if (type === 'init') {
    console.log('here is the first load of stuff', stuff);
  } else {
    console.log('and here is the new stuff', stuff);
  }
});
```

`type` can be `init` or `update` (and potentially other exotic things in future? who knows?)

An exciting discovery: we can do `$inspect({ ...stuff }, console.trace)`.

Reply:
That sounds good to me. Adding a new possible value to an enum like this would technically be a breaking change - but I think we're okay handwaving that away, because this will just be used for debugging in dev builds.